azext_iot/operations/hub.py

```diff
@@ -163,11 +163,43 @@ def _create_self_signed_cert(subject, valid_days, output_path=None):
     return create_self_signed_certificate(subject, valid_days, output_path)
 
 
+def update_iot_device_custom(instance, edge_enabled=None, status=None, status_reason=None,
+                             auth_method=None, primary_thumbprint=None, secondary_thumbprint=None,
+                             primary_key=None, secondary_key=None):
+    if edge_enabled is not None:
+        instance['capabilities']['iotEdge'] = edge_enabled
+    if status is not None:
+        instance['status'] = status
+    if status_reason is not None:
+        instance['statusReason'] = status_reason
+    if auth_method is not None:
+        if auth_method == DeviceAuthType.shared_private_key.name:
+            auth = 'sas'
+            if (primary_key and not secondary_key) or (not primary_key and secondary_key):
+                raise CLIError("primary + secondary Key required with sas auth")
+            instance['authentication']['symmetricKey']['primaryKey'] = primary_key
+            instance['authentication']['symmetricKey']['secondaryKey'] = secondary_key
+        elif auth_method == DeviceAuthType.x509_thumbprint.name:
+            auth = 'selfSigned'
+            if not any([primary_thumbprint, secondary_thumbprint]):
+                raise CLIError("primary + secondary Thumbprint required with selfSigned auth")
+            instance['authentication']['x509Thumbprint']['primaryThumbprint'] = primary_thumbprint
```

Comment:
There is an issue here if primary or secondary thumbprint is set (but not the other), then one of them becomes None -- basically deleting the existing value for the property.